tests/RPCServer.test.ts

```diff
@@ -58,423 +58,358 @@ describe(`${RPCServer.name}`, () => {
-    await expect(errorEventP).toReject();
-    await rpcServer.stop({ force: true });
-  });
+      }
+      const rpcServer = new RPCServer({ logger, idGen });
+      await rpcServer.start({
+        manifest: {
+          testMethod: new TestMethod({}),
+        },
+      });
+      const [outputResult, outputStream] =
+        rpcTestUtils.streamToArray<Uint8Array>();
+      let activeStream: PromiseCancellable<void> | undefined = undefined;
+      const tapStream = rpcTestUtils.tapTransformStream<Uint8Array>(
+        async (_, iteration) => {
+          if (iteration === 2) {
+            // @ts-ignore: kidnap private property
+            const activeStreams = rpcServer.activeStreams.values();
+            for (const activeStream_ of activeStreams) {
+              activeStream = activeStream_;
+              activeStream_.cancel(Error('Some error'));
+            }
+          }
+        },
+      );
+      void tapStream.readable.pipeTo(outputStream).catch(() => {});
+      const readWriteStream: RPCStream<Uint8Array, Uint8Array> = {
+        cancel: () => {},
+        readable: stream,
+        writable: tapStream.writable,
+      };
+      rpcServer.handleStream(readWriteStream);
+      const result = await outputResult;
+      const lastMessage = result[result.length - 1];
+      expect(activeStream).toBeDefined();
+      await expect(activeStream!).toResolve();
+      expect(lastMessage).toBeDefined();
+      expect(() =>
+        rpcUtils.parseJSONRPCResponseFailed(JSON.parse(lastMessage.toString())),
+      ).not.toThrow();
+      await rpcServer.stop({ force: true });
+    },
+  );
+  test.prop({ messages: specificMessageArb })(
+    'handler can yield nothing',
+    async ({ messages }) => {
+      const stream = rpcTestUtils.messagesToReadableStream(messages);
+      class TestMethod extends DuplexHandler {
+        public handle = async function* (
+          input: AsyncGenerator<JSONRPCRequestParams>,
+          _cancel: (reason?: any) => void,
+          _meta: Record<string, JSONValue> | undefined,
+          _ctx: ContextTimed,
+        ): AsyncGenerator<JSONRPCResponseResult> {
+          // Consume values
+          for await (const _ of input);
+        };
+      }
+      const rpcServer = new RPCServer({ logger, idGen });
+      await rpcServer.start({
+        manifest: {
+          testMethod: new TestMethod({}),
+        },
+      });
+      const [outputResult, outputStream] = rpcTestUtils.streamToArray();
+      const readWriteStream: RPCStream<Uint8Array, Uint8Array> = {
+        cancel: () => {},
+        readable: stream,
+        writable: outputStream,
+      };
+      rpcServer.handleStream(readWriteStream);
+      await outputResult;
+      // We're just expecting no errors
+      await rpcServer.stop({ force: true });
+    },
+  );
   test.prop({
     messages: specificMessageArb,
     error: rpcTestUtils.errorArb(rpcTestUtils.errorArb()),
   })(
-    'should send error message with sensitive',
```

Comment:
This test was also removed because it was basically the same as the test without sensitive. I'm not exactly sure what 'sensitive' means in this context, but from the outside, the test seemed redundant.

Reply:
The sensitive is used to ensure no sensitive info is leaked via the error call, but in this case, the tests weren't testing anything proper, so the tests themselves can be removed.